### PR TITLE
all: import individual symbols feature

### PR DESCRIPTION
**Pull request by ryan-willis**
Implements the `fn` and `struct` import features from #5747 

- Clarify the options available in the docs
- `v fmt` opts for shortest syntax based on the `import` statement (basic)
- Added AST sub-statements for error/warning output accuracy
- Various tests for error messages and usage options

### CODE REVIEW

**Comment by Delta456** [comment]
```v
assert term.white('INFO') == white('INFO')
```

I think we shouldn't allow this because we are specifically importing `white`  from `term` module so it should be just `white`. If you do this in other language which have this then it gives an error. Eg: How Python does it 

```python
>>> from math import pi
>>> pi
3.141592653589793
>>> math.pi
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'math' is not defined
```
But you can do `math.pi` if you do `import math`. This needs a lot of discussions I believe.

**Comment by ryan-willis** [comment]
@Delta456 thanks! I agree that it _should_ just be `white` (which `v fmt` will do for you). I only wrote the tests to prove that the feature is working, and I'm not attached to having `term` imported along with its members.

I'll leave the decision to restrict it up to the community 👍

File: vlib/v/parser/fn.v [hunk 1]
```diff
@@ -81,10 +81,18 @@ pub fn (mut p Parser) call_expr(language table.Language, mod string) ast.CallExp
 		p.next()
 		or_kind = .propagate
 	}
+	mut fn_mod := p.mod
+	if registered := p.table.find_fn(fn_name) {
+		if registered.is_placeholder {
+			p.table.fns.delete(fn_name)
```
Comment (spytheman) [inline vlib/v/parser/fn.v:87]: why is this needed?

**Comment by spytheman** [comment]
I think the current implementation is fine. Doing `import term` and `import term { white }` should produce identical results, except that in the second case, `white` should be available without its `term.` prefix *too*, which was the desired effect in the issue.

**Comment by spytheman** [comment]
Plus testing that it works by doing a direct comparison is very nice.

File: vlib/v/parser/fn.v (hunk 1 above)
Comment (ryan-willis) [reply vlib/v/parser/fn.v]: Good question; it isn't needed. Must be kruft I neglected to take out. It's gone now

**Comment by medvednikov** [comment]
Great work! Thanks.

**Comment by mindplay-dk** [comment]
> I think the current implementation is fine. Doing `import term` and `import term { white }` should produce identical results, except that in the second case, `white` should be available without its `term.` prefix _too_, which was the desired effect in the issue.

I have to disagree.

If I'm trying to import only certain specific symbols, importing the namespace symbol itself could be a problem - for example, if it has a name that I'd wanted to use for something else, now I can't.

With the current behavior of automatically importing both symbols, what happens in the case where the namespace name is identical to the symbol name?

    import leftpad { leftpad }

No other language I can think of will automatically import both.

**Comment by ryan-willis** [comment]
@mindplay-dk when the namespace name and the symbol name is identical, you can still import and use both. The V parser will recognize which symbol is being referred to by the context of its use.

If you really wanted to use the module as a name (I'm not sure why you'd want that since that would introduce ambiguity by itself), you can always alias it.

Your snippet works, as does:
```
import leftpad as lp { leftpad }
```

**Comment by mindplay-dk** [comment]
> If you really wanted to use the module name as another name in the code (I'm not sure why you'd want that since that would introduce ambiguity by itself), you can always alias it.

In my optic, this is unnecessary learning curve - in the spirit of keeping the language simple, I'd really like to see *one* import statement resulting in *one* imported symbol.

Besides requiring no workarounds, this is generally what other languages do - so it would be a lot less surprising, too.
